torch_xla/csrc/aten_xla_bridge.cpp

```diff
@@ -119,8 +119,12 @@ void XlaUpdateTensors(
     tensorflow::gtl::ArraySlice<const at::Tensor> source_cpu_tensors,
     tensorflow::gtl::ArraySlice<const size_t> indices) {
   for (auto index : indices) {
-    XLATensor xtensor = GetXlaTensorUnwrap(dest_xla_tensors.at(index));
-    xtensor.UpdateFromTensor(source_cpu_tensors.at(index));
+    if (dest_xla_tensors.at(index).device().is_cpu()) {
```

Comment:
This looks not needed ... unless `self` is a CPU tensor in `resize_as_()`, which would be super screwy, to be called into XLA with an in place op which has `self` being another tensor type.
If `self` is not a CPU tensor, then in `resize_as_()` is the only "writable" tensor, so the indices only lists it.

Reply:
Here:
```C++
at::Tensor & AtenXlaTypeDefault::resize_as_(at::Tensor & self, const at::Tensor & the_template) {
  XLA_COUNTER("aten::resize_as_", 1);
  TF_VLOG(3) << "XLA resize_as_ :" << " self=" << self.toString() << " the_template=" << the_template.toString();
  std::vector<at::Tensor> xlatens_tensors = {self, the_template};
  auto xlatens = bridge::XlaCreateTensorList(xlatens_tensors);
  auto&& x_result = at::resize_as_(xlatens[0], xlatens[1]);
  std::vector<size_t> xlatens_update_indices = {0};
  bridge::XlaUpdateTensors(xlatens_tensors, xlatens, xlatens_update_indices);
  static_cast<void>(x_result); // Avoid warnings in case not used
  return self;
}
```
Only `self` will be updated, and `self` it gotta be an XLA tensor.

Reply:
https://github.com/pytorch/pytorch/blob/master/test/test_torch.py#L7041 gives a CPU `self` :(

Reply:
Note that we also let it go as it is for CPU tensors when try creating XLA tensor list. https://github.com/pytorch/xla/blob/master/torch_xla/csrc/aten_xla_bridge.cpp#L98

Reply:
That is different. On the inbound side, we need to be able to handle mixed CPU/XLA tensors.
Here is about being called with `self` on an in place op, being a CPU tensor.
Here the *subject* of the op is a CPU tensor?
What is the logic there?

How many other surprises like that the new multiple dispatch is going to gift us?
Because if now all the ops that we handle/lower can start coming in with mixed tensor types, it is going to be hell.
What about a convolution with input on CPU and kernel in XLA?
Do we get that too?